File: legacy/leet-121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        ans = 0
        minPrev = prices[0]
        for i in range(1, len(prices)):
            ans = max(ans ,prices[i] - minPrev)
            minPrev = min(minPrev, prices[i])
        return ans
        

File: Leet121.java
class Solution {
    public int maxProfit(int[] prices) {
        int curMax = prices[prices.length-1];
        int ans = 0;
        for (int i = prices.length - 2; i >= 0; i--) {
            ans = Math.max(ans, curMax-prices[i]);
            curMax = Math.max(curMax, prices[i]);
        }

        return ans;
    }
}